src/DETHRACE/common/car.c

```diff
@@ -5054,65 +4947,36 @@ int CollideCamera2(br_vector3* car_pos, br_vector3* cam_pos, br_vector3* old_cam
-                    face_list[1].v[0].v[2] = face_list[2].v[0].v[2];
-                    sb = face_list[2].normal.v[2] * face_list[0].normal.v[2]
-                        + face_list[2].normal.v[1] * face_list[0].normal.v[1]
-                        + face_list[2].normal.v[0] * face_list[0].normal.v[0];
+                sb = BrVector3Dot(&face_list[0].normal, &face_list[1].normal);
+                if (sb > 0.95f && k > 2) {
+
+                    face_list[1].normal = face_list[2].normal;
+                    face_list[1].v[0] = face_list[2].v[0];
+                    sb = BrVector3Dot(&face_list[2].normal, &face_list[0].normal);
                     k = 2;
                 }
-                if (sb <= 0.94999999) {
-                    tv2.v[0] = cam_pos->v[0] - face_list[1].v[0].v[0];
-                    tv2.v[1] = cam_pos->v[1] - face_list[1].v[0].v[1];
-                    tv2.v[2] = cam_pos->v[2] - face_list[1].v[0].v[2];
-                    sc = face_list[1].normal.v[2] * tv2.v[2]
-                        + face_list[1].normal.v[1] * tv2.v[1]
-                        + face_list[1].normal.v[0] * tv2.v[0];
+                if (sb <= 0.95f) {
+                    BrVector3Sub(&tv2, cam_pos, &face_list[1].v[0]);
+                    sc = BrVector3Dot(&face_list[1].normal, &tv2);
                     if (sc < hither && sc >= 0.0) {
-                        sc = face_list[1].normal.v[2] * face_list[0].normal.v[2]
-                            + face_list[1].normal.v[1] * face_list[0].normal.v[1]
-                            + face_list[1].normal.v[0] * face_list[0].normal.v[0];
-                        b.v[0] = face_list[0].normal.v[0] * sc;
-                        b.v[1] = face_list[0].normal.v[1] * sc;
-                        b.v[2] = face_list[0].normal.v[2] * sc;
-                        face_list[1].normal.v[0] = face_list[1].normal.v[0] - b.v[0];
-                        face_list[1].normal.v[1] = face_list[1].normal.v[1] - b.v[1];
-                        face_list[1].normal.v[2] = face_list[1].normal.v[2] - b.v[2];
-                        tv2.v[0] = (hither - ts2) * face_list[1].normal.v[0];
-                        tv2.v[1] = (hither - ts2) * face_list[1].normal.v[1];
-                        tv2.v[2] = (hither - ts2) * face_list[1].normal.v[2];
-                        cam_pos->v[0] = cam_pos->v[0] + tv2.v[0];
-                        cam_pos->v[1] = cam_pos->v[1] + tv2.v[1];
-                        cam_pos->v[2] = cam_pos->v[2] + tv2.v[2];
+                        sc = BrVector3Dot(&face_list[0].normal, &face_list[1].normal);
+                        BrVector3Scale(&b, &face_list[0].normal, sc);
+                        BrVector3Sub(&face_list[1].normal, &face_list[1].normal, &b);
+                        BrVector3Scale(&tv2, &face_list[1].normal, hither - ts2);
+                        BrVector3Accumulate(cam_pos, &tv2);
                     }
                 }
             }
         }
-        i += k;
+        i += k + 1;
```

Comment:
where did this `+1` come from? I don't see that in mine?

Reply:
which is the part of this function that you thought might have been misplaced?

Reply:
The original code was correct.
The `+1` comes from the `for` post-increment.
I was looking at a `while` implementation of the loop which combined the `+k` and `+1`.